centreon/src/Core/Notification/Infrastructure/Repository/DbServiceGroupResourceRepository.php

```diff
@@ -0,0 +1,397 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Notification\Infrastructure\Repository;
+
+use Centreon\Domain\Log\LoggerTrait;
+use Centreon\Infrastructure\DatabaseConnection;
+use Core\Common\Infrastructure\Repository\AbstractRepositoryRDB;
+use Core\Notification\Application\Converter\NotificationServiceEventConverter;
+use Core\Notification\Application\Repository\NotificationResourceRepositoryInterface;
+use Core\Notification\Domain\Model\NotificationGenericObject;
+use Core\Notification\Domain\Model\NotificationResource;
+use Core\Notification\Domain\Model\NotificationServiceEvent;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
+use Utility\SqlConcatenator;
+
+class DbServiceGroupResourceRepository extends AbstractRepositoryRDB implements NotificationResourceRepositoryInterface
+{
+    use LoggerTrait;
+
+    private const RESOURCE_TYPE = 'servicegroup';
+    private const EVENT_ENUM = NotificationServiceEvent::class;
+    private const EVENT_ENUM_CONVERTER = NotificationServiceEventConverter::class;
+
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function supportResourceType(string $resourceType): bool
+    {
+        return mb_strtolower($resourceType) === self::RESOURCE_TYPE;
```

Comment:
The interface argument is `$type`, not `$resourceType`
```suggestion
    public function supportResourceType(string $type): bool
    {
        return mb_strtolower($type) === self::RESOURCE_TYPE;
```